src/Circuit.java

```diff
@@ -0,0 +1,234 @@
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+
+/**
+ *
+ *
+ * @author Caden Hansen
+ */
+public final class Circuit {
+
+    /**
+     * Instance variables (fields)
+     */
+
+    /**
+     * arrayList to represent Circuit.
+     */
+    private ArrayList<Integer> circuit;
+
+    /**
+     * arrayList of arrayLists to allow for parallel circuits.
+     *
+     * not fully fleshed out rn, all the methods will likely need to be changed
+     * in someway to account for potential parallelCircuits but I think the
+     * utility would allow for more depth inside of the component. The ability
+     * to add comparators and other such circuit parts.
+     *
+     */
+    private ArrayList<ArrayList<Integer>> parallelCircuit;
```

Comment:
I don't know enough about circuits to guide you here, but maybe you can model individual circuit parts (i.e., resistors, capacitors, etc.). Modeling a complete circuit seems challenging, and a modular approach might be more appropriate. That said, I will leave it up to you to decide what makes the most sense.